### PR TITLE
iotop-c: add new package

### PR DESCRIPTION
**Pull request by arezvov**
###### Motivation for this change

It's just an alternative version of iotop package written in C.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/iotop-c/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    make install DESTDIR=$out
+    ln -s $out/usr/sbin $out/bin
+    mv $out/usr/sbin/iotop $out/usr/sbin/${pname}
+  '';
+
+  meta = with lib; {
+    description = "iotop identifies processes that use high amount of input/output requests on your machine";
+    homepage = "https://github.com/Tomas-M/iotop";
+    license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iotop-c/default.nix:25]: Missing maintainer and is the license gpl2Only or gpl2Plus?

Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME``.

File: pkgs/os-specific/linux/iotop-c/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "v1.17";
+
+  src = fetchFromGitHub {
+    owner = "Tomas-M";
+    repo = "iotop";
+    rev = version;
+    sha256 = "0hjy30155c3nijx3jgyn5kpj293632p0j6f3lf5acdfax1ynav86";
+  };
+
+  buildInputs = [ ncurses pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iotop-c/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
  buildInputs = [ ncurses ];
```

File: pkgs/os-specific/linux/iotop-c/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  buildPhase = "make DESTDIR=$out";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iotop-c/default.nix:15]:
```suggestion
  makeFlags = [ "DESTDIR=$out" ];
```

File: pkgs/os-specific/linux/iotop-c/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  installPhase = ''
+    make install DESTDIR=$out
+    ln -s $out/usr/sbin $out/bin
+    mv $out/usr/sbin/iotop $out/usr/sbin/${pname}
+  '';
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iotop-c/default.nix:20]:
```suggestion
  installFlags = [ "DESTDIR=$out" ];

  postInstall = ''
    mv $out/usr/sbin/iotop $out/bin/iotop-c
  '';
```

nixpkgs usually do not use sbin or usr directories.

File: pkgs/os-specific/linux/iotop-c/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{stdenv, fetchFromGitHub, lib, ncurses, pkg-config }:
+
+stdenv.mkDerivation rec {
+  pname = "iotop-c";
+  version = "v1.17";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iotop-c/default.nix:5]:
```suggestion
  version = "1.17";
```

File: pkgs/os-specific/linux/iotop-c/default.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Tomas-M";
+    repo = "iotop";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/iotop-c/default.nix:10]:
```suggestion
    rev = "v${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126697` at 6c3a9ac6 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/68b5b66e26f9c6feae971fae5448c1c2">iotop-c</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by arezvov** [comment]
@SuperSandro2000 thank you. I applied your suggestions.

**Pull request merged**